pkg/client/dtclient/config_client.go

```diff
@@ -132,9 +132,17 @@ func (d *ConfigClient) ExistsWithName(ctx context.Context, api api.API, name str
 }
 
 func (d *ConfigClient) UpsertByName(ctx context.Context, a api.API, name string, payload []byte) (entity DynatraceEntity, err error) {
-	if a.ID == api.Extension {
+	if isExtension(a) {
 		return d.uploadExtension(ctx, a, name, payload)
 	}
+
+	if isSlo(a) {
+		if isSloV1, parseErr := isSloV1Payload(payload); parseErr != nil {
```

Comment:
At first, glance hard to read since the method is `isSloV1Payload`, and I would expect that I would go into the if it is, but then I see that an error is returned. Then, there is another if for the payload is V1. 

I would probably refactor it to something that the `isSloV1Payload` would only return an error. The detail about the error being a parse error or a validation error the code here doesn't care it would return in any case...

```
err := isSloV1Payload(payload)
if isSlo(a) && err != nil {
   return DynatraceEntity{}, err
}
```

Reply:
I'll keep the validate check still inside the isSlo block, because we don't need to validate each and every payload.
Will also rename it to validateSloV1Payload because "is..." would implicitly tell that we have a boolean value